flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/source/reader/RowConverter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.source.reader;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.conversion.DataStructureConverter;
+import org.apache.flink.table.data.conversion.DataStructureConverters;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.types.utils.TypeConversions;
+import org.apache.flink.types.Row;
+import org.apache.iceberg.flink.FlinkSchemaUtil;
+import org.jetbrains.annotations.NotNull;
+
+public class RowConverter implements RowDataConverter<Row> {
```

Comment:
code probably can be simplified. see example from `RowDataToRowMapper`

```
  @Override
  public void open(Configuration parameters) throws Exception {
    this.converter =
        DataStructureConverters.getConverter(TypeConversions.fromLogicalToDataType(rowType));
  }

  @Override
  public Row map(RowData value) throws Exception {
    return (Row) converter.toExternal(value);
  }
}
```

Reply:
thanks for the feedback. Do you mean change this to RowData To Row mapper?

Reply:
I meant refer to the `RowDataToRowMapper` for simpler implementation of this converter.

Reply:
@stevenzwu simplified the code. Can you take a look? If it looks good then i can make changes to the test cases.

Currently, there’s no utility function to extract RowTypeInfo from RowType, so I have to use the RowTypeInfo class to create it. Not sure if this is the correct way but let me know if there's a better way. Thanks

Reply:
LGTM